sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/PartitioningAwareFileIndex.scala

```diff
@@ -69,7 +75,7 @@ abstract class PartitioningAwareFileIndex(
           val files: Seq[FileStatus] = leafDirToChildrenFiles.get(path) match {
             case Some(existingDir) =>
               // Directory has children files in it, return them
-              existingDir.filter(isNonEmptyFile)
+              existingDir.filter(f => isNonEmptyFile(f) && matchGlobPattern(f))
```

Comment:
Nice, @gengliangwang . BTW, can we switch put `matchGlobPattern(f)` first like `matchGlobPattern(f) && isNonEmptyFile(f)` in order to avoid `f.getLen` more?